From ee73ae586a307aad04f00908122f3950303153db Mon Sep 17 00:00:00 2001
From: Christian Persch <chpe@gnome.org>
Date: Wed, 5 Oct 2011 20:07:09 +0200
Subject: [PATCH] Just call g_type_init

This will init gthreads, so no need to do it manually.
---
 src/lib/ar-runtime.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/src/lib/ar-runtime.c b/src/lib/ar-runtime.c
index eeff9f9..f51766f 100644
--- a/src/lib/ar-runtime.c
+++ b/src/lib/ar-runtime.c
@@ -286,17 +286,9 @@ ar_runtime_init (const char *name)
   }
 #endif /* G_OS_WIN32 */
 
-#if defined(HAVE_GNOME) || defined(HAVE_RSVG_GNOMEVFS) || defined(ENABLE_SOUND)
-  /* If we're going to use gconf, gnome-vfs, or canberra, we need to
-   * init threads; and this has to be done before calling any other glib functions.
-   */
-#if defined(LIBGAMES_SUPPORT_GI)
-  /* Seed has already called g_thread_init() */
-  g_assert (g_thread_get_initialized());
-#else
-  g_thread_init (NULL);
-#endif
-#endif
+  /* This also initialises gthread */
+  g_type_init ();
+
   /* May call any glib function after this point */
 
   ar_profilestart ("ar_runtime_init");
-- 
1.7.12

